src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.TransactionExecutor.cs

```diff
@@ -116,10 +116,11 @@ private static UInt256 GetResultGas(Transaction transaction, CallOutput result)
                 long gas = result.GasSpent;
                 if (result.AccessList is not null)
                 {
+                    // TODO: this looks wrong after EIP7623 - have to revisit this
                     // if we generated access list, we need to fix actual gas cost, as all storage was considered warm
-                    gas -= IntrinsicGasCalculator.Calculate(transaction, Berlin.Instance);
+                    gas -= IntrinsicGasCalculator.Calculate(transaction, Berlin.Instance).IntrinsicGas;
                     transaction.AccessList = result.AccessList;
-                    gas += IntrinsicGasCalculator.Calculate(transaction, Berlin.Instance);
+                    gas += IntrinsicGasCalculator.Calculate(transaction, Berlin.Instance).IntrinsicGas;
```

Comment:
i think it is a bit more complicated than that, if i understand what we are doing here correctly.
floorGas before and after the accessList will be the same
so if `gas > floorGas` then everything is okay, we can go ahead with the current logic as with accessList added, the cost will only increase
but if `gas == floorGas`, then I think we have no way of calculating the correct gas except for executing the transaction again, because in this case the actual gas spent < floorGas, that is why we used floorGas. But now we dont know the actualGasSpent and we cant know the value of actualGasSpent + accessListCost

Reply:
Is it we should compute the used gas as before 7623 first, then compare with the floor after the transaction execution to determine the actual gas to be used? I have similar wonder when I look into geth's pull request.
https://github.com/ethereum/go-ethereum/pull/29040
I feel like the eip is a bit confusing to me.

Reply:
i think the EIP is clear, its just this specific case, that has some issues.